docs/sections/learn/steps/index.md

```diff
@@ -0,0 +1,134 @@
+next(conversation_template.process([{"instruction": "Hello", "response": "Hi"}]))
+# [
+#   {
+#     "instruction": "Hello",
+#     "response": "Hi",
+#     "conversation": [
+#       {
+#         "role": "user",
+#         "content": "Hello"
+#       },
+#       {
+#         "role": "assistant",
+#         "content": "Hi"
+#       }
+#     ]
+#   }
+# ]
+```
+
+It takes the dictionary with data, adds another `conversation` with the data formatted as a conversation template, and passes this data to the following step.
+
+This is a small type step that shows what to expect when we are creating our `Step` objects, which can start from something as simple as generating a conversation template from some columns on a dataset.
+
+## step decorator
+
+If all that we want to apply in a step is some simple processing, it can be easier to just create a plain function, and decorate it. We can find more examples in the [API reference][distilabel.steps.decorator], but let's see how we could define the previous step as a function and use the decorator:
+
+```python
+from distilabel.steps.decorator import step
+from distilabel.steps.typing import StepOutput
+from distilabel.steps.base import StepInput
+
+# Normal step
+@step(inputs=["instruction", "response"], outputs=["conversation"])
+def ConversationTemplate(inputs: StepInput) -> StepOutput:
+    for input in inputs:
+        input["conversation"] = [
+            {"role": "user", "content": input["instruction"]},
+            {"role": "assistant", "content": input["response"]},
+        ]
+    yield inputs
+```
+
+Which can be instantiated exactly the same as the `ConversationTemplate` class:
+
+```python
+conversation_template = ConversationTemplate(
+    name="my-conversation-template",
+    pipeline=Pipeline(name="my-pipeline"),
+)
+```
+
+This `@step` decorator has a special type depending `step_type` which will be better understood once we see the different types of steps.
+
+## Types of steps
+
+Other than the general or normal steps we have seen, there are special types of steps that have a restricted behaviour compared to the general `Step`.
+
+### Generator steps
+
+These are steps that are able to generate data, and don't need to receive any input from previous step, as it's implied in the normal steps. The typical use for these steps will be loading data for example, as can be seen in [`LoadDataFromDicts`][distilabel.steps.generators.data]. For this type of steps we will only need to define the `process` method, and we can optionally pass a `batch_size` argument, that will determine the batch size of the generated batches.
+
+### Global steps
+
+Other special type of step are the global steps. These steps don't have any `inputs` or `outputs`, and their `process` method receives all the data at once instead of using batches. This kind of behavior is necessary for example to push a dataset to a specific place, or doing some filtering on the whole data before continuing with the pipeline.
```

Comment:
```suggestion
Other special type of step are the global steps. The only difference with respect to the `Step` is that their `process` method receives all the data at once instead of using batches. This kind of behaviour is necessary for example to push a dataset to a specific place, or doing some filtering on the whole data before continuing with the pipeline.
```